### ISSUE TITLE
Constraint lost in .lp file

### ISSUE THREAD

**Issue by jajhall** [ERGO-Code/HiGHS#852]
The following `lp_solve` `.lp` file comes from [http://lpsolve.sourceforge.net/5.0/lp-format.htm]

> min: 2 + x1 + 3 + x2 + 4;
> myrow: x1 + x2 >= 2;
> /* comment */
> x1 >= 1;
> x2 >= 1;
> int x1,x2;

The optimal solution is `[1, 1]` with optimal objective value 11

Converting this to `CPLEX` `.lp` raises a few issues. A possible file is

> Minimize
> obj: 2 + x1 + 3 + x2 + 4
> Subject To
> myrow: x1 + x2 >= 2
> /* comment */
> #Bounds
> x1 >= 1
> x2 >= 1
> General
> x1 x2

However, the `/* comment */` line causes 

> Running HiGHS 1.2.2 [date: 2022-05-09, git hash: e09f5a79]
> Copyright (c) 2022 ERGO-Code under MIT licence terms
> terminate called after throwing an instance of 'std::length_error'
>   what():  basic_string::_M_create
> Aborted (core dumped)

_So lp_solve style comments should be recognised and ignored_

Deleting the comment produces

> Running HiGHS 1.2.2 [date: 2022-05-09, git hash: e09f5a79]
> Copyright (c) 2022 ERGO-Code under MIT licence terms
> MIP  cplex has 3 rows; 2 cols; 3 nonzeros; 2 integer variables
> Presolving model
> Presolve: Infeasible

This is because the last constraint is interpreted as `0 >= 1`.: the HighsLp has 3 nonzeros, corresponding to the values in the first two constraints. 

Uncommenting `#Bounds` produces

> Running HiGHS 1.2.2 [date: 2022-05-09, git hash: e09f5a79]
> Copyright (c) 2022 ERGO-Code under MIT licence terms
> MIP  cplex has 1 rows; 2 cols; 2 nonzeros; 2 integer variables
> Presolving model
> 0 rows, 0 cols, 0 nonzeros
> 0 rows, 0 cols, 0 nonzeros
> Presolve: Optimal

and yields the correct solution, _but the optimal objective value is 6, not 11_. 

Clearly each constant term is used to redefine the offset, rather than being added to the previous offset.

**Comment by feldmeier** [comment]
Interesting finds!
the lp_solve comments shouldn't take long to fix.
The 2nd issue I'm not sure about, I'll dig into it.
The third one is doable, but takes a bit of time to implement (a similar issue should be that variable coefficients don't add up if it appears twice in a constraint)

**Comment by jajhall** [comment]
Thanks, and I'm sure you're right about constant terms in constraints. 

I find it very hard to follow what's going on in the .lp file reader, otherwise I'd have had a go at the easier things myself.

**Comment by feldmeier** [comment]
Is the "#" in "#Bounds" intentional?

**Comment by jajhall** [comment]
> Is the "#" in "#Bounds" intentional?

It means that there's no Bounds section, exposing the issue where the last "bound" is interpreted as a constraint

> 0 >= 1
